docs-ui/components/code.tsx

```diff
@@ -0,0 +1,154 @@
+import {withTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+import copy from 'copy-text-to-clipboard';
+import Prism from 'prismjs';
+import 'prismjs/themes/prism.css';
+import {Component, createRef, Fragment, RefObject} from 'react';
+
+import {IconCode} from 'app/icons';
+import space from 'app/styles/space';
+import {Theme} from 'app/utils/theme';
+
+type Props = {
+  theme: Theme;
+  /**
+   * Main code content gets passed as the children prop
+   */
+  children: string;
+  /**
+   * Auto-generated class name for <pre> and <code> element,
+   * with a 'language-' prefix, e.g. language-css
+   */
+  className?: string;
+  /**
+   *  Meta props from the markdown syntax,
+   *  for example, in
+   *
+   * ```jsx label=hello
+   * [some code]
+   * ```
+   *
+   * the label prop is set to 'hello'
+   */
+  label?: string;
+};
+
+type State = {};
+
+class Code extends Component<Props, State> {
```

Comment:
Should these be written with functional components + hooks now?

Reply:
Yepp, will update it. 
Initially wrote this with the intention to add some interactive features that will require a state. But it seems that many of those features are already available in Storybook preview.

Reply:
You can also easily do state using hooks these days! `useState`